apps/sms/views/conversation/js/attachment.js

```diff
@@ -73,32 +73,37 @@
       //   - Blob mimetype is unsupported but file extension is valid.
       //   - File extenion is missing or invalid but mimetype is supported.
 
-      var mimetype =
-        MimeMapper.guessTypeFromFileProperties(
-          this.name,
-          this.blob.type.toLowerCase()
-        );
-      var filename = MimeMapper.ensureFilenameMatchesType(this.name, mimetype);
-
-      // Override filename, so that every attachment that is saved via "open"
-      // activity will be placed in the single location.
-      filename = ATTACHMENT_FOLDER_PATH + getBaseName(filename);
-
-      var activity = new MozActivity({
-        name: 'open',
-        data: {
-          type: mimetype,
-          filename: filename,
-          blob: this.blob,
-          allowSave: options && options.allowSave
-        }
+      navigator.mozL10n.formatValue('unnamed-attachment').then(unnamed => {
```

Comment:
It will force all the attachment viewing localize the unnamed-attachment string, but not all the attachments will need this step. So how about having a promise like this:
```
var l10nPromise = this.name ? Promise.resolve() :
  navigator.mozL10n.formatValue('unnamed-attachment');

return l10nPromise.then(() => {
  ...
});
```
BTW you can also treat activity as promise since it's DomRequest and catch the error together in l10nPromise reject. But it's optional and you might need to replace the mock activity in attachment unit test.